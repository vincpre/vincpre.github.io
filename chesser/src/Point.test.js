/* eslint-env jest */
const Point = require("./Point");

describe("Point", () => {
  describe("distance entre 2 points", () => {
    it("en Y de 1 case", () => {
      const point = new Point(1, 2);
      expect(point.distance(new Point(1, 3))).toMatchSnapshot();
    });
  });
  describe("distance entre 2 points", () => {
    it("en Y de 2 cases", () => {
      const point = new Point(1, 2);
      expect(point.distance(new Point(1, 4))).toMatchSnapshot();
    });
  });
  describe("distance entre 2 points", () => {
    it("en Y de 3 cases", () => {
      const point = new Point(1, 2);
      expect(point.distance(new Point(1, 5))).toMatchSnapshot();
    });
  });
  describe("distance entre 2 points", () => {
    it("en Y de 4 cases et en x de 1 cases", () => {
      const point = new Point(1, 2);
      expect(point.distance(new Point(2, 6))).toMatchSnapshot();
    });
  });
});
