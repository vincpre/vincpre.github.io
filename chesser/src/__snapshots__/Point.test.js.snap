// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point distance entre 2 points en Y de 1 case 1`] = `
Object {
  "x": 0,
  "y": 1,
}
`;

exports[`Point distance entre 2 points en Y de 2 cases 1`] = `
Object {
  "x": 0,
  "y": 2,
}
`;

exports[`Point distance entre 2 points en Y de 3 cases 1`] = `
Object {
  "x": 0,
  "y": 3,
}
`;

exports[`Point distance entre 2 points en Y de 4 cases et en x de 1 cases 1`] = `
Object {
  "x": 1,
  "y": 4,
}
`;
